examples/typescript-interop/node/demo.c

```diff
@@ -0,0 +1,329 @@
+    // so we must override this every time we call this function (as opposed to, say,
+    // setting it once during init).
+    napi_global_env = env;
+
+    // Get the argument passed to the Node function
+    size_t argc = 1;
+    napi_value argv[1];
+
+    status = napi_get_cb_info(env, info, &argc, argv, NULL, NULL);
+
+    if (status != napi_ok)
+    {
+        return NULL;
+    }
+
+    napi_value node_arg = argv[0];
+
+    struct RocStr roc_arg;
+
+    // Write the length of the Node string into `roc_arg_len`
+    size_t roc_arg_len;
+
+    status = napi_get_value_string_utf8(env, node_arg, NULL, 0, &roc_arg_len);
+
+    if (status != napi_ok)
+    {
+        return NULL;
+    }
+
+    // Node always writes a null terminator, so we need to keep that in mind.
+    size_t nul_terminated_str_bytes = roc_arg_len + 1;
+
+    // Create a RocStr from the Node string
+    size_t roc_arg_strlen;
+    char* roc_arg_bytes;
+
+    if (nul_terminated_str_bytes < sizeof(struct RocStr))
+    {
+        // If it can fit in a small string, use the string itself as the buffer.
+        roc_arg_bytes = (char*)&roc_arg;
+    }
+    else
+    {
+        // It was too big for a small string, so do a heap allocation and write into that.
+        roc_arg_bytes = (char*)roc_alloc(nul_terminated_str_bytes, __alignof__(char));
+    }
+
+    status = napi_get_value_string_utf8(env, node_arg, roc_arg_bytes, nul_terminated_str_bytes, &roc_arg_strlen);
+
+    if (status != napi_ok)
+    {
+        return NULL;
+    }
+
+    struct RocStr roc_str_arg = init_rocstr((uint8_t*)roc_arg_bytes, roc_arg_strlen);
+
+    // Call the Roc function to populate `roc_ret`'s bytes.
+    struct RocStr roc_ret;
+
+    printf("Calling Roc...\n"); // TODO small string is busted for some reason here
+
+    // for(size_t i = 0; i < roc_arg_strlen; i++) {
+    //     printf("%c ", ((char*)&roc_str_arg)[i]);
+    // }
+    // printf("\n");
+
+    roc__mainForHost_1_exposed_generic(&roc_ret, &roc_str_arg);
+
+    printf("Called Roc...\n");
+
+    // Create a Node string from the Roc string and return it.
+    char* roc_str_contents;
+
+    if (is_small_str(roc_ret))
+    {
+        // In a small string, the string itself contains its contents.
+        roc_str_contents = (char*)&roc_ret;
+    }
+    else
+    {
+        roc_str_contents = (char*)roc_ret.bytes;
+    }
+
+    napi_value node_ret;
+
+    status = napi_create_string_utf8(env, roc_str_contents, roc_str_len(roc_ret), &node_ret);
+
+    if (status != napi_ok)
+    {
+        return NULL;
+    }
+
+    return node_ret;
+}
+
+napi_value init(napi_env env, napi_value exports) {
+    napi_status status;
+    napi_value fn;
+
+    status = napi_create_function(env, NULL, 0, call_roc, NULL, &fn);
+
+    if (status != napi_ok)
+    {
+        return NULL;
+    }
+
+    status = napi_set_named_property(env, exports, "hello", fn);
+
+    if (status != napi_ok)
+    {
+        return NULL;
+    }
+
+    return exports;
```

Comment:
You never decref and free the data returned from roc. It will leak.

Reply:
Great catch! Fixed in 6f8cfc62299be9debf5f6d95d9a358c4cb03e0fc